reference/src/layout/unions.md

```diff
@@ -6,29 +6,109 @@ not to change until an RFC ratifies them.
 
 [#13]: https://github.com/rust-rfcs/unsafe-code-guidelines/issues/13
 
-The only degree of freedom the compiler has when computing the layout of a union
-like
+### Layout of individual union fields
+
+A union consists of several variants, one for each field. All variants have the
+same size and start at the same memory address, such that in memory the variants
+overlap. This can be visualized as follows:
 
 ```rust,ignore
-union U { f1: T1, f2: T2 }
+[ <--> [field0_ty] <----> ]
+[ <----> [field1_ty] <--> ]
+[ <---> [field2_ty] <---> ]
 ```
 
-is to determine the offset of the fields. The layout of these fields themselves
-is already entirely determined by their types, and since we intend to allow
-creating references to fields (`&u.f1`), unions do not have any wiggle-room
-there.
+> **Figure: union field layout**: Each row in the picture shows the layout of
+> the union for each of its fields, where the square brackets `[]` depict an
+> array of bytes, `<-...->` denotes different amount of padding, and
+> `[field{i}_ty]` is the bytes of the type of the `i`-th union field.
```

Comment:
The figure caption here has different style than what you suggested in https://github.com/rust-lang/unsafe-code-guidelines/pull/186. I think I prefer the other style, could you adjust this here?